contracts/collateralUpdater/CollateralRequirementUpdater.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity ^0.4.24;
+
+import "../Agreement.sol";
+import "./PriceOracle.sol";
+
+contract CollateralRequirementUpdater {
+
+    uint256 constant public MAX_DISPUTABLE_APPS = 10;
+
+    Agreement public agreement;
+    DisputableAragonApp[] public disputableApps;
+    ERC20[] public collateralTokens;
+    uint256[] public actionAmountsStable;
+    uint256[] public challengeAmountsStable;
+    PriceOracle public priceOracle;
+    address public stableToken;
+
+    constructor(
+        Agreement _agreement,
+        DisputableAragonApp[] _disputableApps,
+        ERC20[] _collateralTokens,
+        uint256[] _actionAmountsStable,
+        uint256[] _challengeAmountsStable,
+        PriceOracle _priceOracle,
+        address _stableToken
+    ) public {
+        require(_disputableApps.length == _collateralTokens.length
+            && _actionAmountsStable.length == _challengeAmountsStable.length
+            && _collateralTokens.length == _actionAmountsStable.length, "ERROR: Inconsistently sized arrays");
+        require(_disputableApps.length <= MAX_DISPUTABLE_APPS, "ERROR: Too many disputable apps");
+
+        agreement = _agreement;
+        disputableApps = _disputableApps;
+        collateralTokens = _collateralTokens;
+        actionAmountsStable = _actionAmountsStable;
+        challengeAmountsStable = _challengeAmountsStable;
+        priceOracle = _priceOracle;
+        stableToken = _stableToken;
+    }
+
+    // This contract requires the MANAGE_DISPUTABLE_ROLE permission on the specified Agreement contract
+    function updateCollateralRequirements() external {
+        for (uint256 i = 0; i < disputableApps.length; i++) {
+            DisputableAragonApp disputableAragonApp = disputableApps[i];
+            (, uint256 currentCollateralRequirementId) = agreement.getDisputableInfo(disputableAragonApp);
+            (ERC20 collateralToken, uint64 challengeDuration, uint256 actionAmount, uint256 challengeAmount)
+                = agreement.getCollateralRequirement(disputableAragonApp, currentCollateralRequirementId);
+
+            require(collateralToken == collateralTokens[i], "ERROR: Collateral tokens do not match");
```

Comment:
I'm pretty sure it's needed. We don't want to update the collateral requirements with an incorrect price if the token has been changed through Agreements as you've pointed out. It wouldn't necessarily revert without this require statement.

Yeah we could make it Ownable and add a setter. Will do if I have time/energy.

Reply:
> I'm pretty sure it's needed. We don't want to update the collateral requirements with an incorrect price if the token has been changed through Agreements as you've pointed out.

That's true, didn't think of it this way.

If for example one of the disputable Aragon apps collateral token would change, it would be necessary to deploy a new instance of this contract correct?